sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/Rules/UnusedPrivateMemberTest.cs

```diff
@@ -161,58 +158,90 @@ public class AssetPostprocessor { }
-                new CS.UnusedPrivateMember(),
-                ParseOptionsHelper.FromCSharp8,
-                additionalReferences: NuGetMetadataReference.NETStandardV2_1_0);
-        }
+                                    new CS.UnusedPrivateMember(),
+                                    ParseOptionsHelper.FromCSharp8,
+                                    additionalReferences: NuGetMetadataReference.NETStandardV2_1_0);
 
         [TestMethod]
         [TestCategory("CodeFix")]
-        public void UnusedPrivateMember_CodeFix()
+        public void UnusedPrivateMember_CodeFix() =>
+            Verifier.VerifyCodeFix(@"TestCases\UnusedPrivateMember.cs",
+                                   @"TestCases\UnusedPrivateMember.Fixed.cs",
+                                   @"TestCases\UnusedPrivateMember.Fixed.Batch.cs",
+                                   new CS.UnusedPrivateMember(),
+                                   new CS.UnusedPrivateMemberCodeFixProvider());
+
+        [TestMethod]
+        [TestCategory("Rule")]
+        public void UnusedPrivateMember_UsedInGeneratedFile() =>
+            Verifier.VerifyAnalyzer(new[]
+                                    {
+                                        @"TestCases\UnusedPrivateMember.CalledFromGenerated.cs",
+                                        @"TestCases\UnusedPrivateMember.Generated.cs"
+                                    },
+                                    new CS.UnusedPrivateMember());
+
+        [TestMethod]
+        [TestCategory("Performance")]
+        public void UnusedPrivateMember_Performance()
         {
-            Verifier.VerifyCodeFix(
-                @"TestCases\UnusedPrivateMember.cs",
-                @"TestCases\UnusedPrivateMember.Fixed.cs",
-                @"TestCases\UnusedPrivateMember.Fixed.Batch.cs",
-                new CS.UnusedPrivateMember(),
-                new CS.UnusedPrivateMemberCodeFixProvider());
+            Action verifyAnalyzer = () => Verifier.VerifyAnalyzer(new[] {@"TestCases\UnusedPrivateMember.Performance.cs"},
+                                                                  new CS.UnusedPrivateMember(),
+                                                                  additionalReferences:
+                                                                  GetEntityFrameworkCoreReferences(Constants.NuGetLatestVersion));
+
+            // Once the NuGet packages are downloaded, the time to execute the analyzer on the given file is
+            // about ~1 sec. It was reduced from ~11 min by skipping Guids when processing ObjectCreationExpression.
+            // The threshold is set here to 30 seconds to avoid flaky builds due to slow build agents or network connections.
+            verifyAnalyzer.ExecutionTime().Should().BeLessOrEqualTo(30.Seconds());
         }
 
         [TestMethod]
-        [TestCategory("Rule")]
-        public void UnusedPrivateMember_UsedInGeneratedFile()
+        public void UnusedPrivateMember_GuidFP()
```

Comment:
I can't put this inside the `UnusedPrivateMember.cs` with the other test cases because the codefix will remove the constructor and the file will not compile.

Reply:
shouldn't this test better be inside `UnusedPrivateMemberTest.Types.cs` ?